README.md

```diff
@@ -1,3 +1,67 @@
-# Authentication spinner component
+# Authentication spinner
 
-Production code for the spinner component used by the Authentication Team
+## About The Project
+
+This spinner component is initialised on a web page rendered by the authentication frontend app (but owned by the Orchestration Team) to replace a server-rendered form with content and interactive controls that update to reflect the readiness state returned by an API.
+
+States represented by the spinner component include:
+
+1. Pending
+2. Error
+3. Long wait
+4. Completed
+
+### How the authentication spinner works
+
+#### 1. Getting a reference to the required DOM element
+
+When the component JavaScript runs in a user's browser, a DOM lookup is performed for an element with the id of `spinner-container`. This element is required as a mounting point for the component. If the DOM element is not found, the spinner will not run and the user can proceed to use the non-JavaScript version.
+
+#### 2. Initialising the spinner content
+
+Where this element is found, it is passed as an argument to the `Spinner` constructor. The constructor will then try to populate the `Spinner` content attribute using `data-*` attributes found on the DOM element included in the server-rendered page (this approach has been taken so that content and language translations can be managed alongside other content in authentication frontend).
```

Comment:
Are there parameters that someone setting this up needs to pass through to the spinner? If so, they might be good to include here / somewhere?

Reply:
Thanks Becka - that's a good point. At the moment, there aren't any parameters to be passed because the `spinner-app.js` [file](https://github.com/govuk-one-login/authentication-spinner/blob/main/src/spinner-app.js) initialises itself with the two parameters (the string for the DOM lookup and the element that's passed to the constructor. 

I was thinking of this as being simplest for our individual use case because there's no need to introduce any additional JavaScript for the spinner to work for us (we simply need to include and defer the script from the module in the page). I anticipate this will need to change when it's consumed by other teams and/or added to the Frontend Capability Team monorepo but I'm expecting quite a lot to need to be parameterised at that point. I've been trying to think of the spinner as being introduced in discrete stages:

1. We do some work to produce and document something that works for our specific use case
2. There's a piece of work (perhaps conducted by Frontend Capability, perhaps by us or another team) to extend and parameterise the component so it can be used by multiple teams
3. Ongoing maintenance of this component within the Frontend Capability Team monorepo. 

All that said, I'd be happy to tweak the approach at this stage so we do pass a parameter in. Let me know what you think.

One thing I have done in the meantime is to reorder the README so that "Getting started" is earlier and include more explanation along with an example of the expected HTML structure and elements